cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftService.java

```diff
@@ -0,0 +1,127 @@
+   * @return List of {@link Draft}
+   * @throws RuntimeException when an error occurs while fetching from the table
+   */
+  public List<Draft> listDrafts(NamespaceSummary namespaceSummary, String owner,
+                                boolean includeConfig, SortRequest sortRequest,
+                                @Nullable String filter) {
+    List<Draft> drafts = store.listDrafts(namespaceSummary, owner, sortRequest, includeConfig);
+    if (!Strings.isNullOrEmpty(filter)) {
+      drafts = drafts.stream()
+        .filter(draft -> draft.getName().toLowerCase().startsWith(filter.toLowerCase()))
+        .collect(Collectors.toList());
+    }
+    return drafts;
+  }
+
+  /**
+   * Fetch the given draft
+   *
+   * @param draftId {@link DraftId} that is used to uniquely identify a draft
+   * @return the {@link Draft} object
+   * @throws RuntimeException when an error occurs while fetching from the table
+   * @throws DraftNotFoundException if the draft does not exist
+   */
+  public Draft getDraft(DraftId draftId) throws RuntimeException, DraftNotFoundException {
+    return store.getDraft(draftId).orElseThrow(() -> new DraftNotFoundException(draftId));
+  }
+
+  /**
+   * Write the given draft
+   *
+   * @param draftId {@link DraftId} that is used to uniquely identify a draft
+   * @param draftStoreRequest {@link DraftStoreRequest} that contains the rest of the draft data
+   * @throws RuntimeException when an error occurs while writing to the table
+   */
+  public <T extends ETLConfig> void writeDraft(DraftId draftId,
+                                               DraftStoreRequest<T> draftStoreRequest) {
+    // TODO(CDAP-17456): Add collision detection using the hashes
+    store.writeDraft(draftId, draftStoreRequest);
+
+    metrics.gauge(Constants.Metrics.DRAFT_COUNT, store.getDraftCount());
+  }
+
+  /**
+   * Delete the given draft
+   *
+   * @param draftId {@link DraftId} that is used to uniquely identify a draft
+   * @throws RuntimeException when an error occurs while fetching from the table
+   * @throws DraftNotFoundException if the draft does not exist
+   */
+  public void deleteDraft(DraftId draftId) {
+    // Make sure the draft exists before attempting to delete it
+    getDraft(draftId);
+    store.deleteDraft(draftId);
+
+    metrics.gauge(Constants.Metrics.DRAFT_COUNT, store.getDraftCount());
+  }
+
+  /**
+   * Checks if the given field exists in the {@link DraftStore} table spec. This should be used for validation.
+   *
+   * @param fieldName name of the field to check
+   * @return True if the field exists in the {@link DraftStore} table spec
+   */
+  public boolean fieldExists(String fieldName) {
+    Map<String, FieldType.Type> fieldTypes = DraftStore.TABLE_SPEC
+      .getFieldTypes()
+      .stream()
+      .collect(Collectors.toMap(FieldType::getName, FieldType::getType));
+    return fieldTypes.containsKey(fieldName);
```

Comment:
Use filter and find first to check instead of creating a map.

Reply:
I removed the map and used `anyMatch` to get the answer directly since we just want a true/false value.